tests/publiccloud/azure_cli.pm

```diff
@@ -74,6 +81,14 @@ sub cleanup {
     return 1;
 }
 
+sub post_run_hook {
+    cleanup();
+}
+
+sub post_fail_hook {
```

Comment:
AFAIK `post_run_hook` is also called when test is failed . can you please add a VR with inserting `die ` into random place of `run`  function . I bet you will got cleanup called twice